include/uapi/sound/sof/abi.h

```diff
@@ -59,4 +61,34 @@
 /* SOF ABI magic number "SOF\0". */
 #define SOF_ABI_MAGIC		0x00464F53
 
+#define SOF_MANIFEST_DATA_TYPE_NHLT 1
+
+/**
+ * struct sof_manifest_tlv - SOF manifest TLV data
+ * @type: type of data
+ * @size: data size (not including the size of this struct)
+ * @data: payload data
+ */
+struct sof_manifest_tlv {
+	__u32 type;
+	__u32 size;
+	__u8 data[];
+};
+
+/**
+ * struct sof_manifest - SOF topology manifest
+ * @abi_major: Major ABI version
+ * @abi_minor: Minor ABI version
+ * @abi_patch: ABI patch
+ * @count: count of tlv items
+ * @items: consecutive variable size tlv items
+ */
+struct sof_manifest {
+	__u8 abi_major;
+	__u8 abi_minor;
+	__u8 abi_patch;
```

Comment:
fyi: The firmware uses u16 fort he version numbers.

Reply:
We add the same version number as the firmware in topology as well. I wonder why it is only 3 bytes. @juimonen any idea?

Reply:
as example fw version: `Booted firmware version: 10.29.0.5399` 5399 is 'build' number in major.minor.hotfix.build

Reply:
@ujfalusi this is with the cavs FW right? the tplg is built with the SOF repo and it will be of the format major.minor.patch

Reply:
Yes, it was with the cavs FW, but the fields are u16 for both (and SOF also have the dot build in version.

Reply:
right I think we made a mistake with the ABI in topology by using only 1 byte. fixing it now would mean new tplg will fail with older kernel.

Reply:
@ujfalusi i've fixed the size of the ABI bytes in manifest now with https://github.com/thesofproject/sof/pull/5631. This will also need an update to @juimonen alsautils PR.